lacommunaute/utils/tests/tests_utils.py

```diff
@@ -535,3 +535,12 @@ def test_size_validator(self, db):
         with pytest.raises(Exception):
             file.file.size = 1024 * 1024 * 5 + 1
             file.save()
+
+
+class ContextlessTemplateTestCase(TestCase):
+    @patch("django.template.Template.__init__", ContextlessTemplate.__init__)
+    def test_template_rendering(self):
+        template = Template("<p>{{ variable }}</p>")
+        context = Context({"variable": "test"})
+        rendered = template.render(context)
+        assert "{{ variable }}" in rendered
```

Comment:
@vincentporte ajouté dans la nouvelle version pour faciliter les testes avec les variables de la contexte... on peut ensuite l'utiliser dans nos testes à la manière de : 

```python
    @patch('django.template.Template.__init__', ContextlessTemplate.__init__)
    def test_template_documentation(self, client, db, snapshot):
        forum = CategoryForumFactory(
            with_public_perms=True, with_child=True
        ).children.first()
        response = client.get(
            reverse(
                "forum_extension:forum",
                kwargs={"pk": forum.pk, "slug": forum.slug}
            )
        )
        content = parse_response_to_soup(response, replace_in_href=[forum])

        upvotes_area = content.find_all("div", id=re.compile("^upvotesarea"))

        assert len(upvotes_area) == 1
        assert str(upvotes_area[0]) == snapshot(name="template_documentation_upvotes")
```

Les fichiers `.ambr` resultants sont dans la forme du gabarit avant la contexte est appliquée, mais où les expressions e.g. `{% for %}`, `{% if %}` sont bien appliqués, e.g.

```html
<div class="d-inline-block" id="upvotesarea{{ obj.pk }}">
```

qu'est-ce que tu penses ?

Reply:
Discuté aujourd'hui et on va remplacer cette solution avec quelque chose similaire que le Factory.Trait `for_snapshot` de les-emplois